File: Interviews/Striver-SDE-Sheet/Day-4-Arrays/Longest-Substring-Without-Repeating-Characters/3.cpp
int lengthOfLongestSubstring(string s) {
    vector<int> index(100, -1);
    int start = 0, ans = 0, n = s.size();
    for(int i = 0; i < n; i++){
        if(index[s[i] - ' '] != -1) start = max(start, index[s[i] - ' '] + 1);
        ans = max(ans, i - start + 1);
        index[s[i] - ' '] = i;
    }
    
    return ans;
}

File: Interviews/Placement-Preparation/Day158-03January2023/codes/3.java
public class LC519 {
    int start, n, m;
    public LC519(int m, int n) {
        start = 0;
        this.n = m;
        this.m = n;
    }

    public int[] flip() {
        int[] ans = {start / m, start % m};
        start = (start + 1) % (n * m);
        return ans;
    }

    public void reset() {
    }
}